### PR TITLE
Created authgroups checking

### LINKED ISSUE #8: Auth groups
## Auth middleware for different groups
What needs to be implemented:
- Through middleware:
  - [ ] Checking different auth groups
- In app:
  - [ ] Model for user groups
  - [ ] Linking of user to group

### PR DESCRIPTION
**Pull request by antoniomika**
Closes #8
### PR Description
Created authgroups checking

### CODE REVIEW

File: server/db/model/User.js [hunk 1]
```diff
@@ -307,6 +307,20 @@ schema.methods.sendPasswordResetEmail = function() {
     });
 };
 
+schema.methods.checkGroup = function(checkGroup) {
+    if (checkGroup === 'all') {
```
Comment (connorkrupp) [inline server/db/model/User.js]: what does this part do

Comment (connorkrupp) [inline server/db/model/User.js]: Like what does 'all' mean

**Comment by connorkrupp** [comment]
Thats a lot of parameters in those functions 0.0

File: server/db/model/User.js (hunk 1 above)
Comment (antoniomika) [inline server/db/model/User.js]: The user doesn't necessarily need to have a group, but if they are in a group, any of them should work. Don't really know if we need that check, but maybe `any` is better

**Comment by antoniomika** [comment]
Yeah, probably should break that stuff up into more functions

**Comment by antoniomika** [comment]
Came up with a solution that I think is both pretty clean and works much better. Let me know what you think.

File: server/db/index.js [hunk 2]
```diff
@@ -7,7 +7,9 @@ mongoose.Promise = global.Promise;
 mongoose
     .connect('mongodb://' + config.mongo_hostname + '/' + config.backend_db)
     .then(res => {
-        console.log(res);
+        if (res) {
+            console.log(res);
+        }
```
Comment (connorkrupp) [inline server/db/index.js]: YES thank you lol

Comment (connorkrupp) [inline server/db/index.js]: I was wondering where that 'undefined' was coming from

File: server/db/model/User.js [hunk 3]
```diff
@@ -307,6 +307,32 @@ schema.methods.sendPasswordResetEmail = function() {
     });
 };
 
+schema.methods.checkGroup = function(checkGroup) {
+    if (checkGroup === 'any') {
```
Comment (connorkrupp) [inline server/db/model/User.js]: yeah thats perfect

**Comment by connorkrupp** [comment]
👍

**Comment by connorkrupp** [comment]
I'm gonna make an issue for documenting how to use auth groups in a route / creating new auth groups, but feel free to merge @antoniomika

**Pull request merged**